### PR TITLE
Convert all metadata columns to JSONB

### LINKED ISSUE #18: `metadata` storage is not safe
The current `metadata` storage is not safe.

Currently, key and value is `_` separated and it's easily breakable if we have `_` in the `key`.
My suggestion is to use JSONB for the DB type and use `serde` to (de)serialize it.

### PR DESCRIPTION
**Pull request by omid**
FIXES #18

### CODE REVIEW

File: scheduler/crates/api/Cargo.toml [hunk 1]
```diff
@@ -11,11 +11,10 @@ nettu_scheduler_api_structs = { path = "../api_structs" }
 nettu_scheduler_domain = { path = "../domain" }
 nettu_scheduler_infra = { path = "../infra" }
 serde = { version = "1.0", features = ["derive"] }
-futures = "0.3" 
+futures = "0.3"
 actix-web = "3.3.2"
 actix-cors = "0.5.4"
 async-trait = "0.1.42"
-env_logger = "0.8.2"
```
Comment (omid) [inline scheduler/crates/api/Cargo.toml:18]: This was not in use! (next PR I will run `cargo update` :P)

File: scheduler/crates/api/src/event/create_event.rs [hunk 2]
```diff
@@ -219,7 +219,7 @@ mod test {
             busy: false,
             calendar_id: calendar.id.clone(),
             user,
-            reminders: vec![],
```
Comment (omid) [inline scheduler/crates/api/src/event/create_event.rs:222]: macros will slow down compile time, where there is an equivalent, it's better to use that one.

File: scheduler/crates/infra/src/repos/schedule/postgres.rs [hunk 3]
```diff
@@ -93,7 +75,7 @@ impl IScheduleRepo for PostgresScheduleRepo {
             schedule.id.inner_ref(),
             Json(&schedule.rules) as _,
             schedule.timezone.to_string(),
-            &metadata
+            Json(&schedule.metadata) as _,
```
Comment (omid) [inline scheduler/crates/infra/src/repos/schedule/postgres.rs:78]: These kind of `Json` doesn't work without the `as _`. What's the secret?

**Comment by omid** [comment]
@fmeringdal I still don't know how can I run the tests locally, could you please put the commands inside the `Makefile` I provided in the root of the project?

**Comment by fmeringdal** [comment]
@omid I added test command to makefile now, let me know if it works for you.
I also added a test action when a PR is created, updated so it is possible to see if tests are passing before merging PR. You can try it out if make a new dummy commit to this PR

File: scheduler/crates/infra/src/repos/schedule/postgres.rs (hunk 3 above)
Comment (fmeringdal) [reply scheduler/crates/infra/src/repos/schedule/postgres.rs:78]: I am not really sure about the exact reason. I just noted that this is how they do it in the sqlx examples: https://github.com/launchbadge/sqlx/blob/master/examples/postgres/json/src/main.rs#L68

**Review by fmeringdal — APPROVED** [review]
Quality work as always. Thanks!

**Pull request merged**